addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlHandler.java

```diff
@@ -0,0 +1,235 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Niko Home Control: ACTIONID does not match an action in the controller " + actionId);
+        }
+    }
+
+    private void handleSwitchCommand(Integer actionId, Command command) {
+        if (command instanceof OnOffType) {
+            OnOffType s = (OnOffType) command;
+            if (s == OnOffType.OFF) {
+                nhcComm.executeAction(actionId, 0);
+            } else {
+                nhcComm.executeAction(actionId, 100);
+            }
+        }
+    }
+
+    private void handleBrightnessCommand(Integer actionId, Command command) {
+        if (command instanceof OnOffType) {
+            OnOffType s = (OnOffType) command;
+            if (s == OnOffType.OFF) {
+                nhcComm.executeAction(actionId, 255);
+            } else {
+                nhcComm.executeAction(actionId, 254);
+            }
+        } else if (command instanceof IncreaseDecreaseType) {
+            IncreaseDecreaseType s = (IncreaseDecreaseType) command;
+            int stepValue = ((Number) this.getConfig().get(CONFIG_STEP_VALUE)).intValue();
+            int currentValue = nhcComm.getActionState(actionId);
+            int newValue;
+            if (s == IncreaseDecreaseType.INCREASE) {
+                newValue = currentValue + stepValue;
+                // round down to step multiple
+                newValue = newValue - newValue % stepValue;
+                nhcComm.executeAction(actionId, (newValue > 100 ? 100 : newValue));
+            } else {
+                newValue = currentValue - stepValue;
+                // round up to step multiple
+                newValue = newValue + newValue % stepValue;
+                nhcComm.executeAction(actionId, (newValue < 0 ? 0 : newValue));
+            }
+        } else if (command instanceof PercentType) {
+            PercentType p = (PercentType) command;
+            nhcComm.executeAction(actionId, p.intValue());
+        }
+    }
+
+    private void handleRollershutterCommand(Integer actionId, Command command) {
+        if (command instanceof UpDownType) {
+            UpDownType s = (UpDownType) command;
+            if (s == UpDownType.UP) {
+                nhcComm.executeAction(actionId, 255);
+                logger.debug("Niko Home Control: rollershutter {} up pressed", actionId);
+            } else {
+                nhcComm.executeAction(actionId, 254);
+                logger.debug("Niko Home Control: rollershutter {} down pressed", actionId);
+            }
+        } else if (command instanceof StopMoveType) {
+            StopMoveType s = (StopMoveType) command;
+            if (s == StopMoveType.STOP) {
+                nhcComm.executeAction(actionId, 253);
+                logger.debug("Niko Home Control: rollershutter {} stop pressed", actionId);
+            } else {
+                logger.debug("Niko Home Control: rollershutter {} move pressed", actionId);
```

Comment:
Actually, I did not figure out yet what MOVED is supposed to do exactly yet. As I don't have rollershutters myself, I have been relying on inputs from other people when implementing this. From feedback, rollershutters do work, but I am not sure I have covered all cases yet. The log statement will help me to refine this in the future if needed, based on user feedback.